src/tools/merssh/command.cpp

```diff
@@ -67,6 +69,47 @@ int Command::executeRemoteCommand(const QString &command)
     return rc;
 }
 
+int Command::executeSfdk(const QStringList &arguments)
+{
+    QTextStream qerr(stderr);
+
+    static const QString program = QCoreApplication::applicationDirPath()
+        + '/' + RELATIVE_REVERSE_LIBEXEC_PATH + "/sfdk" QTC_HOST_EXE_SUFFIX;
+
+    QStringList allArguments;
+    if (!targetName().isEmpty())
+        allArguments << "-c" << "target=" + targetName();
+    if (!deviceName().isEmpty())
+        allArguments << "-c" << "device=" + deviceName();
+    allArguments << arguments;
+
+    auto environment = QProcessEnvironment::systemEnvironment();
+    environment.insert(Mer::Constants::SAILFISH_SDK_FRONTEND,
+            Mer::Constants::SAILFISH_SDK_FRONTEND_ID);
+
+    qerr << "+ " << program << ' ' << QtcProcess::joinArgs(allArguments, Utils::OsTypeLinux)
+        << endl;
+
+    QProcess process;
+    process.setProcessChannelMode(QProcess::ForwardedChannels);
+    process.setProgram(program);
+    process.setArguments(allArguments);
+    process.setProcessEnvironment(environment);
+
+    maybeUndoCMakePathMapping();
+
+    int rc{};
+    process.start();
+    if (!process.waitForFinished(-1) || process.error() == QProcess::FailedToStart)
+        rc = -2; // like QProcess::execute does
+    else
+        rc = process.exitStatus() == QProcess::NormalExit ? process.exitCode() : -1;
+
+    maybeDoCMakePathMapping();
```

Comment:
To me it looks like the only additional value merssh provides here is the path mapping. I think it would be good to consider adding that functionality to sfdk and calling sfdk directly. I don't mean that we should do that now, but maybe at some convenient point in the future.

Reply:
Yup, path mapping is definitely going to be moved to sfdk (at some point) but the rest depends on the exact meaning of "here" and "invoking directly". merssh also provides the cached queries and this is something I am not sure we should taint sfdk side with.

Reply:
Please interpret my writings as something that makes sense ;)

I was only thinking about these cases where we use merssh as a thin wrapper for sfdk. So some logic would be needed before calling to determine which binary to call. And by "invoking directly", I was thinking about modifying the wrapper scripts to call sfdk instead of merssh, but of course other possibilities do exist. 

It's really not that straight forward, that's why I wrote that we should not do it now.

Reply:
:)